### ISSUE TITLE
error at save

### ISSUE THREAD

**Issue by PatrickKwinten** [edm00se/AnAppOfIceAndFire#5]
when I try to save a document e.g. from /index.html#/houses/C01D0D19567F92FAC1257E6C0024F7F8 I get the error:

Error: $scope.houseForm[fldNm] is undefined
$scope.saveHouseForm/<@http://dev1/apps/others/rest/AnAppOfIceAndFire.nsf/js/houseApp.js:152:1
r@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:7:300
$scope.saveHouseForm@http://dev1/apps/others/rest/AnAppOfIceAndFire.nsf/js/houseApp.js:151:4
ib.prototype.functionCall/<@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:199:301
Ec[c]</<.compile/</</f@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:216:74
Pe/this.$get</n.prototype.$eval@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:126:11
Pe/this.$get</n.prototype.$apply@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:126:236
Ec[c]</<.compile/</<@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:216:124
x.event.dispatch@http://code.jquery.com/jquery-2.0.3.min.js:5:9954
x.event.add/y.handle@http://code.jquery.com/jquery-2.0.3.min.js:5:6772

**Comment by edm00se** [comment]
Curious, as there's no js file called "houseApp.js" file within the source, or a reference to it in any js file or html file. I did a search of the repo via the GH interface.

Are you using the current version from the master branch?

**Comment by PatrickKwinten** [comment]
Excuse me! I have taken (now) the current version and then I get multiple error messages: Error: input is undefined
@http://dev1/apps/others/java/anAppOfIceAndFire.nsf/js/houseApp.js:189:4
ib.prototype.filter/<@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:194:412
ib.prototype.filter/<@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:194:348
p/c<@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:109:307
Pe/this.$get</n.prototype.$digest@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:123:219
Pe/this.$get</n.prototype.$apply@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:126:291
l@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:81:238
M@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:85:342
wf/</F.onload@http://ajax.googleapis.com/ajax/libs/angularjs/1.3.15/angular.min.js:86:367

**Comment by edm00se** [comment]
Still, it's citing the first line after "Error: input is undefined" as having an error at line 189 of ...nsf/js/houseApp.js. I'll clone a fresh copy of my repo and see how it runs. I'll update either this afternoon or evening.

Also, I assume you are using the full NSF as opposed to the back-end spoofing (via `json-server`).

Also, performing `bower install` should, using the current `master` branch, install Angular at the 1.4.x level, so you shouldn't be loading anything from 1.3.x, unless you've modified things differently.

**Comment by edm00se** [comment]
To confirm, in a DDE environment (with the appropriate `java.pol` edit and inclusion of relevant JARs), I did the following:
1. cloned the `master` branch
2. ran `npm install` and `bower install`
3. symlink'd public to WebContent in the ODP (windows: `mklink /D public ODP\WebContent`, may require `cmd.exe` to "Run as Administrator")
4. ran `gulp build` (the front-end asset build, without the back-end spoof)
5. imported the ODP via Package Explorer
6. created the new NSF from the ODP
7. launched my local web preview of `index.html` in the appropriate app path
8. created, viewed, updated, and deleted some house records
9. observed that my built `style.min.css` and `scripts.js` were building into the root of the ODP, so I edited my `gulpfile.js` to specify the destination for those being w/i `ODP/css` and `ODP/js` respectively (`gulpfile.js` lines # 58 and 65), then re-rean `gulp build`
10. observed a build path error, pointing to the correct `lwpd.domino.adapter.jar` (appears to be version/release number dependent, should only break for local web preview, combined with the fact that my old work PC has Notes + DDE installed in a non-default location); I corrected the path in Build Properties
11. observed an injector error in the angular app (via js console), in the injects to the module for `westerosiApp`

I'll need to update a couple of things in the client-side app, but I'm getting no network errors at least.

Long story short, I'm in the middle of a giant refactor of this project repository, to incorporate my task runner project repo, along with the client-side portion of the app into its own git subtree (issue #4). To view this in its last known good status, you should check out and use [the latest release](/releases/latest) ([v0.5.3.1](https://github.com/edm00se/AnAppOfIceAndFire/tree/0.5.3.1)). That release includes a copy of the [working binary NSF in a zip file](https://github.com/edm00se/AnAppOfIceAndFire/releases/download/0.5.3.1/An.App.of.Ice.and.Fire.zip) (taken from my DDE env at home) and is a better place to start, for now.

**Comment by edm00se** [comment]
The commit level of `master` that should coincide with the zip'd version of the app should be e991d40. That version of things pre-dates the inclusion (part of what I'm bringing back into the main code base) of gulp and its optimizations and injections. The process for cloning the latest release (commit e991d40 or [the pre-built version in the latest release](https://github.com/edm00se/AnAppOfIceAndFire/releases/latest)) is far more straight forward, 
- `npm install`
- symlink the `public` directory to `NSF/WebContent/`
- `bower install`
- `npm start` to spoof the back-end, or directly in the pre-built NSF

**Comment by edm00se** [comment]
I'm going to close this issue on account of:
1. there are a couple of issues known regarding the project structure (and git repo) refactor, those are identified and being worked on, and the app can be cloned or downloaded in its last known good version; see [latest release](https://github.com/edm00se/AnAppOfIceAndFire/releases/latest), or clone at the comit e991d40 (aka- [tag 0.5.3.1](https://github.com/edm00se/AnAppOfIceAndFire/releases/tag/0.5.3.1))
2. I'm still utterly confused as to how you can be getting errors in your browser regarding a version of Angular that the app doesn't use in either its current `master` branch state or the latest release (you're getting errors with links to the 1.3.15 version, when the latest release uses 1.4.3)

I've confirmed that the latest release works as expected, so please take the current "dust" in the `master` branch with a grain of salt. As I move forward, I'll include specific verbiage regarding the recommended path to only download/install the latest release. I've also added a draft release for what will be the next milestone version, 1.1.0 (treating the fully working app as version 1.0, plus-ing up a "feature version" a la semver for the inclusion of the other project contents and assets).
